probes/requiresUpToDateBranches/impl_test.go

```diff
@@ -0,0 +1,202 @@
+	//nolint:govet
+	tests := []struct {
+		name     string
+		raw      *checker.RawResults
+		outcomes []finding.Outcome
+		err      error
+	}{
+		{
+			name: "Requires up to date branches on 1/1 branches",
+			raw: &checker.RawResults{
+				BranchProtectionResults: checker.BranchProtectionsData{
+					Branches: []clients.BranchRef{
+						{
+							Name: &branchVal1,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								CheckRules: clients.StatusChecksRule{
+									UpToDateBeforeMerge: &trueVal,
+								},
+							},
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive,
+			},
+		},
+		{
+			name: "Requires up to date branches on 2/2 branches",
+			raw: &checker.RawResults{
+				BranchProtectionResults: checker.BranchProtectionsData{
+					Branches: []clients.BranchRef{
+						{
+							Name: &branchVal1,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								CheckRules: clients.StatusChecksRule{
+									UpToDateBeforeMerge: &trueVal,
+								},
+							},
+						},
+						{
+							Name: &branchVal2,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								CheckRules: clients.StatusChecksRule{
+									UpToDateBeforeMerge: &trueVal,
+								},
+							},
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive, finding.OutcomePositive,
+			},
+		},
+		{
+			name: "Requires up to date branches on 1/2 branches - 1",
+			raw: &checker.RawResults{
+				BranchProtectionResults: checker.BranchProtectionsData{
+					Branches: []clients.BranchRef{
+						{
+							Name: &branchVal1,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								CheckRules: clients.StatusChecksRule{
+									UpToDateBeforeMerge: &trueVal,
+								},
+							},
+						},
+						{
+							Name: &branchVal2,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								CheckRules: clients.StatusChecksRule{
+									UpToDateBeforeMerge: &falseVal,
+								},
+							},
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive, finding.OutcomeNegative,
+			},
+		},
+		{
+			name: "Requires up to date branches on 1/2 branches - 2",
+			raw: &checker.RawResults{
+				BranchProtectionResults: checker.BranchProtectionsData{
+					Branches: []clients.BranchRef{
+						{
+							Name: &branchVal1,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								CheckRules: clients.StatusChecksRule{
+									UpToDateBeforeMerge: &falseVal,
+								},
+							},
+						},
+						{
+							Name: &branchVal2,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								CheckRules: clients.StatusChecksRule{
+									UpToDateBeforeMerge: &trueVal,
+								},
+							},
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNegative, finding.OutcomePositive,
+			},
+		},
+		{
+			name: "Requires up to date branches on 0/2 branches",
```

Comment:
not avail

Reply:
https://github.com/ossf/scorecard/pull/3691/commits/d7c16773ae85b5c49cffe37a49f8330e60bdd631